src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -296,84 +296,79 @@ public string SanitizeCppVarName(string varName)
             return varName;
         }
 
-        private string CompileSpecialMethod(MethodDesc method, SpecialMethodKind kind)
+        private string CompileSpecialMethod(MethodDesc method)
         {
-            var builder = new CppGenerationBuffer();
-            switch (kind)
+            bool isPInvoke;
```

Comment:
This `bool` makes readability of the code a bit worse (it's non-obvious that the bool being false means it's a runtime import). Maybe we could factor out a new method that takes the original `MethodDesc` and the `importName` string to alleviate that.

But I don't care much. CppWriter is not particularly readable anyway.